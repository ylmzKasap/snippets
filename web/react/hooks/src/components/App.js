import React, { useState } from 'react';
import { EffectComponent } from './UseEffect';
import { ReducerComponent } from './UseReducer';
import { CallbackComponent } from './UseCallback';
import { MemoComponent } from './UseMemo';
import { RefComponent, RefComponentTwo } from './UseRef';
import { CustomHook } from './CustomHook';

const App = () => {
    const [currentDemo, setCurrentDemo] = useState('useEffect');
    const allButtons = [
      'useEffect', 'useReducer', 'useCallback', 'useMemo', 'useRef', 'customHook'];

    return (
      <div id="appContent">
        <div id='buttonContainer'><div className='description'>Other Hooks</div>
          {allButtons.filter(x => x!== currentDemo)
            .map(demo => <button 
              className="nav-button" 
              key={demo}
              onClick={() => setCurrentDemo(demo)}>{demo}</button>)
          }
        </div>
        <div id="demo">
          <div id='currentDemo'>{currentDemo}</div>
          <div className='demoContent'>
            { currentDemo === 'useEffect' && <EffectComponent /> }
            { currentDemo === 'useReducer' && <ReducerComponent /> }
            { currentDemo === 'useCallback' && <CallbackComponent /> }
            { currentDemo === 'useMemo' && <MemoComponent /> }
            { currentDemo === 'useRef' && <div><RefComponent /> <RefComponentTwo /></div> }
            { currentDemo === 'customHook' && <CustomHook /> }
          </div>
        </div>
      </div>
    );
  };
  
  export default App;