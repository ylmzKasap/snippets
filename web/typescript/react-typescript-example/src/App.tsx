import React from "react";
import { Counter } from "./Counter";

// props
// hooks
// render props

const App: React.FC = () => {
  return (
    <div>
      <Counter>
        {({ count, setCount }) => (
          <div>
            {count}
            <button onClick={() => setCount(count + 1)}>+</button>
          </div>
        )}
      </Counter>
    </div>
  );
};

export default App;
