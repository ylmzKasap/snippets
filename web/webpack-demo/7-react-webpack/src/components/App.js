import React from 'react';

const App  = () => {
    return (
        <div>
            <h1>React Template</h1>
        </div>
    )
}

export default App;