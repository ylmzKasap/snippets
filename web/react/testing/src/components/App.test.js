import { getByText, render, screen } from '@testing-library/react';
import App from  '../RouteSwitch';
import userEvent from "@testing-library/user-event";


/* window.fetch = jest.fn(() => {
  const user = { name: 'Jack', email: 'jack@email.com' };

  return Promise.resolve({
    json: () => Promise.resolve(user),
  });
}); */

describe('Renders and . . .',  () => {

  it('Matches the snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('Display the button names correctly', () => {
    render(<App />);
    screen.getByText('Other Hooks');
    screen.getByRole('button', {name: 'useEffect'});
  });

  it('Displays all buttons', () => {
    render(<App />);
    const buttons = screen.getAllByRole('button');
    expect(buttons.length).toEqual(6);
  });

  it('Changes the component', () => {
    render(<App />);
    const customButton = screen.getByRole('button', {name: 'customHook'});
    userEvent.click(customButton);
    expect(global.location.pathname).toEqual('/customHook');
    screen.getByText('Saved Text');
  });
});