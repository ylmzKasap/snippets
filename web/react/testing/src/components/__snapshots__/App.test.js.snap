// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and . . . Matches the snapshot 1`] = `
<div>
  <div
    id="appContent"
  >
    <div
      id="buttonContainer"
    >
      <div
        class="description"
      >
        Other Hooks
      </div>
      <button
        class="nav-button"
      >
        useEffect
      </button>
      <button
        class="nav-button"
      >
        useReducer
      </button>
      <button
        class="nav-button"
      >
        useCallback
      </button>
      <button
        class="nav-button"
      >
        useMemo
      </button>
      <button
        class="nav-button"
      >
        useRef
      </button>
      <button
        class="nav-button"
      >
        customHook
      </button>
    </div>
    <div
      id="demo"
    >
      <div
        id="currentDemo"
      />
      <div
        class="demoContent"
      />
    </div>
  </div>
</div>
`;
